compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3300,8 +3533,14 @@ abstract class GenJSCode extends plugins.PluginComponent
      */
     private def genGlobalJSObject(sym: Symbol)(
         implicit pos: Position): js.Tree = {
-      jsNameOf(sym).split('.').foldLeft(genLoadGlobal()) { (memo, chunk) =>
-        js.JSBracketSelect(memo, js.StringLiteral(chunk))
+      if (isScalaJSDefinedJSClass(sym)) {
+        val classFullName = encodeClassFullName(sym)
+        currentMethodInfoBuilder.addInstantiatedClass(classFullName)
+        js.JSLoadConstructor(jstpe.ClassType(classFullName))
```

Comment:
Consider moving this part to `genPrimitiveJSClass` and assert here that we don't load a Scala.js defined class.

It will give us some safety against trying to load a Scala.js defined class as module.

(maybe also add asserts to the two methods above)...